x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java

```diff
@@ -185,115 +172,73 @@ private static int normalize(int codepoint) {
 
     public static class Defaults {
         public static final FieldType FIELD_TYPE = new FieldType();
-
         static {
             FIELD_TYPE.setTokenized(false);
             FIELD_TYPE.setIndexOptions(IndexOptions.DOCS);
             FIELD_TYPE.setStoreTermVectorOffsets(false);
             FIELD_TYPE.setOmitNorms(true);
             FIELD_TYPE.freeze();
         }
+        public static final TextSearchInfo TEXT_SEARCH_INFO
+            = new TextSearchInfo(FIELD_TYPE, null, Lucene.KEYWORD_ANALYZER, Lucene.KEYWORD_ANALYZER);
         public static final int IGNORE_ABOVE = Integer.MAX_VALUE;
-        public static final String NULL_VALUE = null;
     }
 
-    public static class Builder extends FieldMapper.Builder<Builder> {
-        protected int ignoreAbove = Defaults.IGNORE_ABOVE;
-        protected String nullValue = Defaults.NULL_VALUE;
+    private static WildcardFieldMapper toType(FieldMapper in) {
+        return (WildcardFieldMapper) in;
+    }
 
-        public Builder(String name) {
-            super(name, Defaults.FIELD_TYPE);
-            builder = this;
-        }
+    public static class Builder extends ParametrizedFieldMapper.Builder {
 
-        @Override
-        public Builder docValues(boolean docValues) {
-            if (docValues == false) {
-                throw new MapperParsingException("The field [" + name + "] cannot have doc values = false");
-            }
-            return this;
-        }
+        final Parameter<Integer> ignoreAbove
+            = Parameter.intParam("ignore_above", true, m -> toType(m).ignoreAbove, Defaults.IGNORE_ABOVE)
+            .setValidator(v -> {
+                if (v < 0) {
+                    throw new IllegalArgumentException("[ignore_above] must be positive, got [" + v + "]");
+                }
+            });
+        final Parameter<String> nullValue
+            = Parameter.stringParam("null_value", false, m -> toType(m).nullValue, null).acceptsNull();
 
-        @Override
-        public Builder indexOptions(IndexOptions indexOptions) {
-            if (indexOptions != IndexOptions.DOCS) {
-                throw new MapperParsingException("The field [" + name + "] cannot have indexOptions = " + indexOptions);
-            }
-            return this;
-        }
```

Comment:
Just curious because haven't reviews that many of these moves to ParametrizedFieldMapper yet: These errors woll now happen automatically if no Parameter is defined?

Reply:
Yes, for previously 'global' parameters that don't really make sense on a mapper, you'll get a deprecation warning in 7x and an error in master.